baseplate/context/cassandra.py

```diff
@@ -98,39 +98,34 @@ def make_object_for_context(self, name, span):
         return cqlmapper.connection.Connection(session_adapter)
 
 
-class WaitForCallbackResponseFuture(object):
-    """Wrap the ResponseFuture to ensure callbacks have completed.
+def wrap_future(response_future, callback_fn, callback_args, errback_fn, errback_args):
+    """Patch ResponseFuture.result() to wait for callback or errback to complete.
 
-    The callback_fn and errback_fn passed in the constructor are given
-    special treatment: they must be complete before a result will be
-    returned from result(). They are not given precedence over other
-    callbacks or errbacks, so if another callback triggers the response
-    from the service (and the server span is closed) the special callback might
-    not complete.
+    The callback_fn and errback_fn are given special treatment: they must be
+    complete before a result will be returned from ResponseFuture.result().
+    They are not given precedence over other callbacks or errbacks, so if
```

Comment:
but we can generally rely on getting the first callback in and so finishing first, right?

Reply:
yeah, I'll update the comment to state that